agent/rpc/middleware/interceptors_test.go

```diff
@@ -266,3 +269,23 @@ func TestRequestRecorder(t *testing.T) {
 		})
 	}
 }
+
+func TestGetNetRPCRateLimitingInterceptor(t *testing.T) {
```

Comment:
I think as we are adding tests in here we could add some tests around the interceptor, I'm thinking:
- That `Allow` is called
- That the `Allow` error is returned through the interceptor.
- What would happen if `reqServiceMethod` is not defined in `rpcRateLimitSpecs` (I think that use case is a bug actually) 😃

Reply:
Could be deferred to a separate PR if you think it's better though

Reply:
When the service method is not listed in the rate limit spec then it defaults to "OperationType.READ"